docs/usage.md

```diff
@@ -5,6 +5,24 @@ from the perspective of a user of the tool.
 It may also be useful for BASIL developers as a reference for how to lift
 new binaries for manual testing.
 
+## Running Basil
+
+Basil needs a JVM(>17), recommended JDK-21, which can be installed using system packages.
+
+- e.g. `sudo apt-get install openjdk-21-jdk`
+
+On first run `./mill` (on linux/windows) or `./mill.bat` (on Windows) should bootstrap Scala automatically. Otherwise it can be set up through [coursier](https://get-coursier.io/docs/cli-installation).
+
+### Test Dependencies
+
+Both unit tests and system tests will require a working installation of the Boogie verifier backend and Z3 SMT solver.
+
+These can be installed through [nix](https://github.com/katrinafyi/pac-nix?tab=readme-ov-file#first-time) (recommended):
+
+- `nix profile install nixpkgs#z3 nixpkgs#boogie`
+
+Alternatively, Z3 is usually available in system packages. For manual Boogie install instructions see subheading [Verifying the Boogie file](#verifying-the-boogie-file).
+
 ## Lifting a single binary
```

Comment:
it could be worth adding a breadcrumb link to the docker littering docs, because it's useful for basil Devs to know the difference